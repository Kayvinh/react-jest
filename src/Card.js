import React from "react";
import "./Card.css";

const IMG_BASE = "https://deckofcardsapi.com/static/img";

/** Show a single card.
 *
 * Props:
 * - rank: 1-9,0,J,Q,K,A
 * - suit: C,D,H,S
 */

function Card({ rank, suit }) {
  const id = rank + suit;
  return (
      <img
          className="Card"
          src={`${IMG_BASE}/${id}.png`}
          alt={id}
      />
  );
}
// end

export default Card;
