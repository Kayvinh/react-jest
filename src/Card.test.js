import React from "react";
import { render, screen } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", function () {
  // this is a low-value test, but better than nothing
  render(<Card rank="A" suit="C" />);
});
// end

it("has the correct alt text & src", function () {
  render(<Card rank="A" suit="C" />);

  const cardImage = screen.getByAltText("AC");
  expect(cardImage).toBeInTheDocument();
  expect(cardImage).toHaveAttribute("alt", "AC");
  expect(cardImage).toHaveAttribute(
    "src",
    "https://deckofcardsapi.com/static/img/AC.png"
  );
});
// end

it("has correct class name", function () {
  render(<Card rank="A" suit="C" />);
  const cardImage = screen.getByAltText("AC");
  expect(cardImage).toHaveClass("Card");
});

it("matches snapshot", function () {
  const { container } = render(<Card rank="A" suit="C" />);
  expect(container).toMatchSnapshot();
});
// end
