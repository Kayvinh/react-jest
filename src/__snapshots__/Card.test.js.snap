// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <img
    alt="AC"
    class="Card"
    src="https://deckofcardsapi.com/static/img/AC.png"
  />
</div>
`;
